### PR TITLE
Data and VIS Function loading spinner

### PR DESCRIPTION
**Pull request by saifulkhan**
Test: http://localhost:3000/page?id=60cfad6d81b6034073d2ff16

### CODE REVIEW

**Comment by kachkaev** [comment]
I fixed Netlify in https://github.com/ScottishCovidResponse/rampvis-ui/pull/117

We can merge that PR and push something into your branch, CI should pass

File: src/pages/page.tsx [hunk 1]
```diff
@@ -50,43 +52,55 @@ const PropagatedPage = () => {
...
-      data: dataForVisFunction,
-      links,
-    });
+
+    try {
+      setLoading(true);
+
+      const page = await apiService.get(`/template/page/${pageId}`);
+      console.log("OntoPage: VIS = ", page.vis);
+      console.log("OntoPage: Data = ", page.data);
```
Comment (kachkaev) [inline src/pages/page.tsx:67]: What is our policy on keeping `console.log`? Just wonder if there is a rule that distinguishes accidentally left statements from the ones we want on the main branch.

**Review by kachkaev — APPROVED** [review]
Checked locally – all works 👍

File: src/pages/page.tsx (hunk 1 above)
Comment (saifulkhan) [reply src/pages/page.tsx:67]: Here one special requirement is that VIS developers need  console.log for three variables,e.g, data, vis, and links. Other than that there is no policy. Should we delete other console.log?

Comment (kachkaev) [reply src/pages/page.tsx:67]: Yeah we could. Leaving a temp `console.log` is a very common mistake, so we can even add https://eslint.org/docs/rules/no-console and explicitly disable permitted `console.log`s:

```ts
// eslint-disable-next-line no-console -- VIS developers need....
console.log("this one is allowed");
```

We can do this in a separate PR.

Comment (saifulkhan) [reply src/pages/page.tsx:67]: Thanks

**Pull request merged**